### PR TITLE
Added updates for OpenSSL v3 support

### PR DESCRIPTION
**Pull request by Sreekala-Gopakumar**
https://github.com/eclipse-openj9/openj9-docs/issues/924

Added support for Open SSL 3.0.x (Linux only)
Updated content to mention that OpenSSL 3.0.x is supported for the same algorithms as 1.1.x.
Updated that in case of multiple versions of OpenSSL on the system, the OpenJ9 VM uses the latest version.
Added a note that that OpenSSL 3.0.x does not support initialization vector (IV) sizes above 16Bytes for the GCM algorithm.

[skip ci]

Signed-off-by: Sreekala Gopakumar sreekala.gopakumar@ibm.com

### CODE REVIEW

**Comment by doveye** [comment]
Jenkins doc stage

**Comment by doveye** [comment]
Jenkins doc stage

File: docs/version0.32.md [hunk 1]
```diff
@@ -58,6 +59,12 @@ For compatibility, the following OpenJDK HotSpot options are now supported by Op
 
 You can now use the `SharedClassStatistics` API to get the name, path, and directory of a shared classes cache. Depending on the operating system, you can also get the number of attached VMs for a non-persistent cache. This information is available through the following new methods: `cacheDir()`, `cacheName()`, `cachePath()`, and `numberAttached()`. For more information, see the API documentation.
 
+### Support for OpenSSL 3.0.x
+
+OpenSSL 3.0.x is supported but on Linux only. OpenSSL is enabled by default for the CBC, ChaCha20, ChaCha20-Poly1305Digest, GCM, and RSA cryptographic algorithms.
```
Comment (doveye) [inline docs/version0.32.md:64]: Could use a link to the detailed info on OpenSSL too, eg "For more information about OpenSSL support, see [Cryptographic operations](introduction.md#cryptographic-operations)".

File: docs/version0.32.md [hunk 2]
```diff
@@ -58,6 +59,12 @@ For compatibility, the following OpenJDK HotSpot options are now supported by Op
...
+
+:fontawesome-solid-pencil-alt:{: .note aria-hidden="true"} **Note:** OpenSSL 3.0.x does not support initialization vector (IV) sizes above 16 Bytes for the GCM algorithm. (In earlier OpenSSL versions, you can use such sizes but they might cause unpredictable behavior.) If you need to use a larger size, disable OpenSSL support for the GCM algorithm.
```
Comment (pshipton) [inline docs/version0.32.md:66]: We should refer to the option that does this.

File: docs/jitserver.md [hunk 3]
```diff
@@ -67,7 +67,7 @@ If a JITServer server crashes, the client is forced to perform compilations loca
 
 ## Security
 
-You can encrypt network communication between the client VM and JITServer by using OpenSSL 1.0.x or 1.1.x. To enable encryption, you specify the private key and the certificate at the server and use the certificate at the client. For more information, see [-XX:JITServerSSLCert / -XX:JITServerSSLKey / -XX:JITServerSSLRootCerts](xxjitserversslcert.md).
+You can encrypt network communication between the client VM and JITServer by using OpenSSL 1.0.x, 1.1.x or 3.0.x. To enable encryption, you specify the private key and the certificate at the server and use the certificate at the client. For more information, see [-XX:JITServerSSLCert / -XX:JITServerSSLKey / -XX:JITServerSSLRootCerts](xxjitserversslcert.md).
```
[the remark this answers is not in the document]
Comment (mpirvu) [reply docs/jitserver.md:70]: No, we haven't done anything in that area. I'll test it and get someone to implement support for openssl 3

[the remark this answers is not in the document]
Comment (doveye) [reply docs/jitserver.md:70]: Ok. So it sounds like we should change jitserver.md and xxjitserversslcert.md to say that jitserver doesn't support OpenSSL 3.0.x, and change version0.32.md to say that OpenSSL support is added but only on Linux and not for JITServer.
@mpirvu can you please create a docs issue for changing this text in future when JITServer does support OpenSSL 3.

**Comment by doveye** [comment]
Jenkins doc stage

**Comment by doveye** [comment]
We tried to squash commits but we can't due to merge commits. "Jenkins doc stage" isn't triggering a build so we can't view the updates in the staging doc either.

**Pull request merged**